OlinkAnalyze/R/pca_plot.R

```diff
@@ -14,17 +18,41 @@
 #' @param drop_samples Logical. All samples with any missing values will be dropped.
 #' @param n_loadings Integer. Will plot the top n_loadings based on size.
 #' @param loadings_list Character vector indicating for which OlinkID's to plot as loadings. It is possible to use n_loadings and loadings_list simultaneously.
+#' @param byPanel Perform the PCA per panel (default FALSE)
+#' @param outlierDefX The number standard deviations along the PC plotted on the x-axis that defines an outlier. See also 'Details"
+#' @param outlierDefY The number standard deviations along the PC plotted on the y-axis that defines an outlier. See also 'Details"
+#' @param outlierLines Draw dashed lines at +/-outlierDef[X,Y] standard deviations from the mean of the plotted PCs (default FALSE)
 #' @param verbose Logical. Whether warnings about the number of samples and/or assays dropped or imputed should be printed to the console.
 #' @param ... coloroption passed to specify color order.
-#' @return An object of class "ggplot"
+#' @return A list of objects of class "ggplot"
 #' @keywords NPX, PCA
 #' @export
 #' @examples
 #' \donttest{
 #' library(dplyr)
 #' npx_data <- npx_data1 %>%
 #'     mutate(SampleID = paste(SampleID, "_", Index, sep = ""))
-#' olink_pca_plot(df=npx_data, color_g = "QC_Warning")}
+#'
+#' #PCA using all the data
+#' olink_pca_plot(df=npx_data, color_g = "QC_Warning")
+#'
+#' #PCA per panel
+#' g <- olink_pca_plot(df=npx_data, color_g = "QC_Warning", byPanel = TRUE)
+#' g[[2]] #Plot only the second panel
+#'
+#' #Label outliers
+#' olink_pca_plot(df=npx_data, color_g = "QC_Warning",
+#'                outlierDefX = 4, outlierDefY = 4) #All data
+#' olink_pca_plot(df=npx_data, color_g = "QC_Warning",
+#'                outlierDefX = 4, outlierDefY = 4, byPanel = TRUE) #Per panel
+#'
+#' #Retrieve the outliers
+#' g <- olink_pca_plot(df=npx_data, color_g = "QC_Warning",
+#'                     outlierDefX = 4, outlierDefY = 4, byPanel = TRUE)
+#' outliers <- lapply(g, function(x){x$data}) %>%
```

Comment:
Outliers is empty, maybe change g as well

Reply:
Also reduced example data issue